iceoryx_utils/test/moduletests/test_cxx_forward_list.cpp

```diff
@@ -0,0 +1,1931 @@
+    constexpr uint64_t ELEMENT_COUNT{1};
+    forward_list<CTorTest, CAPACITY> cut;
+    auto iter = cut.before_begin();
+    decltype(CTorTest::m_value) cnt = 0;
+
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(0));
+
+    for (uint64_t i = 0; i < ELEMENT_COUNT; ++i)
+    {
+        iter = cut.emplace_after(iter, cnt);
+        ++cnt;
+    }
+
+    cnt = 0;
+    for (auto& fl : cut)
+    {
+        EXPECT_THAT(fl.m_value, Eq(cnt));
+        ++cnt;
+    }
+
+    EXPECT_THAT(cut.size(), Eq(ELEMENT_COUNT));
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(ELEMENT_COUNT));
+}
+
+TEST_F(forward_list_test, EmplaceAfterWithSomeElements)
+{
+    constexpr uint64_t CAPACITY{42u};
+    constexpr uint64_t ELEMENT_COUNT{37};
+    forward_list<CTorTest, CAPACITY> cut;
+    auto iter = cut.before_begin();
+    uint64_t cnt = 0;
+
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(0));
+
+    for (uint64_t i = 0; i < ELEMENT_COUNT; ++i)
+    {
+        iter = cut.emplace_after(iter, cnt);
+        ++cnt;
+    }
+
+    cnt = 0;
+    for (auto& fl : cut)
+    {
+        EXPECT_THAT(fl.m_value, Eq(cnt));
+        ++cnt;
+    }
+
+    EXPECT_THAT(cut.size(), Eq(ELEMENT_COUNT));
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(ELEMENT_COUNT));
+}
+
+TEST_F(forward_list_test, EmplaceAfterWithCapacityElements)
+{
+    constexpr uint64_t CAPACITY{42u};
+    constexpr uint64_t ELEMENT_COUNT{CAPACITY};
+    forward_list<CTorTest, CAPACITY> cut;
+    auto iter = cut.before_begin();
+    uint64_t cnt = 0;
+
+    for (uint64_t i = 0; i < ELEMENT_COUNT; ++i)
+    {
+        iter = cut.emplace_after(iter, cnt);
+        ++cnt;
+    }
+
+    cnt = 0;
+    for (auto& fl : cut)
+    {
+        EXPECT_THAT(fl.m_value, Eq(cnt));
+        ++cnt;
+    }
+
+    EXPECT_THAT(cut.size(), Eq(ELEMENT_COUNT));
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(ELEMENT_COUNT));
+}
+
+TEST_F(forward_list_test, EmplaceAfterWithMoreThanCapacityElements)
+{
+    constexpr uint64_t CAPACITY{42u};
+    constexpr uint64_t ELEMENT_COUNT{CAPACITY + 1};
+    forward_list<CTorTest, CAPACITY> cut;
+    auto iter = cut.before_begin();
+    uint64_t cnt = 0;
+
+    for (uint64_t i = 0; i < ELEMENT_COUNT; ++i)
+    {
+        iter = cut.emplace_after(iter, cnt);
+        ++cnt;
+    }
+
+    cnt = 0;
+    for (auto& fl : cut)
+    {
+        EXPECT_THAT(fl.m_value, Eq(cnt));
+        ++cnt;
+    }
+
+    EXPECT_THAT(cut.size(), Eq(CAPACITY));
+    EXPECT_THAT(cTor, Eq(0));
+    EXPECT_THAT(customCTor, Eq(CAPACITY));
+}
+
+
+TEST_F(forward_list_test, EmplaceAfterReverseWithOneElements)
+{
+    constexpr uint64_t CAPACITY{42u};
+    constexpr uint64_t ELEMENT_COUNT{1};
+    forward_list<CTorTest, CAPACITY> cut;
+    auto iter = cut.before_begin();
+    uint64_t cnt = 0;
+
+    for (uint64_t i = 0; i < ELEMENT_COUNT; ++i)
+    {
+        cut.emplace_after(iter, cnt);
+        ++cnt;
+    }
+
+    --cnt;
```

Comment:
Very subtle. Would probably have been more visible if in the other test iter did not exist as variable and we called ```sut1.emplace_after(sut1.before_begin(), cnt);```. But this would be inefficient because the compiler would probably evaluate the before_begin function every iteration. It has to, unless it can prove there are no side effects on the list which is not the case, we add elements. It will most likely not spot the fact that before_begin() always returns the same value.

Ok, no changes needed. :-)